### PR TITLE
Add cmake build support to allow building with different tool chains,…

### PR DESCRIPTION
**Pull request by marcschier**
… and make the stack consumable from other cmake based projects that include it as submodule or package.

### CODE REVIEW

**Comment by marcschier** [comment]
Pretty much same changes as PR # 23, just without removing other tool chain files.

**Comment by bernd-edlinger** [comment]
Hi!

Thanks, generally this PR looks acceptable.

Except for the 2016 in the copyright headers.

I have A few questions, though.

- Why are you forcing the --std=c99?
New gcc versions automatically set --std=c11 but it can create interop problems
if different compiler settings are used.  I often miss an easy way to set
additional gcc options in cmake projects, but --std=c99 looks like a candidate
that needs to be controlled by the user.

- How does it work with OpenSSL under windows?
Does it use the openssl libraries from the ./build_winXX.bat?

**Comment by marcschier** [comment]
Hi @bernd-edlinger, 

1. Will update to 2017.
2. I will add a way to override cflags using a cmake option rather than overriding to c99.  
3. On windows cmake finds openssl if it has been installed, e.g. from https://slproweb.com/products/Win32OpenSSL.html.  The logic is at https://github.com/Kitware/CMake/blob/master/Modules/FindOpenSSL.cmake.  Rather than failing I could see if I can honor the $ENV settings from the build_win.bat.

Thanks a lot!

Marc

**Comment by marcschier** [comment]
1. and 2. adjusted, re 3, reading the .bat it appears the openssl install is always under the project root in openssl folder, so I am seeding the search with a hint to that folder.  I have not tested the latter, but can confirm that any installed openssl package is found on both windows and Linux.

Thanks!

Marc

**Comment by bernd-edlinger** [comment]
Sounds good.

Feel free to add a sentence to the ChangeLog.txt.

**Comment by marcschier** [comment]
Done.  Thanks for the review and for taking this change.

**Comment by bernd-edlinger** [comment]
Fine.  Could you squash the 4 commits into one?

**Comment by JuergenKosel** [comment]
Hello @bernd-edlinger,

github does this for you, during the merge of this pull request. If you want this.

File: .gitignore [hunk 1]
```diff
@@ -20,6 +20,7 @@
 */VS2013/*
 */Debug/*
 */Release/*
+build/**
```
Comment (bernd-edlinger) [inline .gitignore]: oh, did you mean "build/*" ?

Comment (marcschier) [inline .gitignore]: No, the intent is to leave out *everything* under build, not just first level.  See https://git-scm.com/docs/gitignore for pattern in git/gitignore.

**Comment by bernd-edlinger** [comment]
Hmm.  Unfortunately now I see the automatic squash done by github did not preserve
Marc's e-mail addess:

```
commit 4a51d28e3c717f8a7f67a459c4c13922cb41a73a
Author: Marc Schier <marcschier@users.noreply.github.com>
Date:   Thu Jun 8 12:51:17 2017 +0200

    Add cmake build support to allow building with different tool chains,… (#48)
    
    - Add cmake build support to allow building with different tool chains, and 
```

Also the commit message looks a bit squashed, sorry for that.

**Pull request merged**